lib/dyph3/differ.rb

```diff
@@ -433,5 +443,72 @@ def self.handle_trailing_newline(ours, base, theirs, result)
         end
         result
       end
+
+      def self.ensure_no_lost_data(left, base, right, return_value)
+        final_result = return_value[2]
+
+        result_word_map = {}
+
+        final_result.each do |result_block|
+          block_text = case result_block[:type]
+            when :non_conflict then result_block[:text]
```

Comment:
@aamontal in a perfect world, yes, but changing that would change the dyph3's API and I didn't want to do that in this PR as it'd break anything using the gem